components/hab/src/command/supportbundle.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2016-2018 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fs::File;
+use std::env;
+use std::path::{Path, MAIN_SEPARATOR};
+use std::process;
+use tar;
+use flate2::Compression;
+use flate2::write::GzEncoder;
+use chrono::Local;
+use error::Result;
+use hcore::fs::FS_ROOT_PATH;
+use common::ui::{Status, UIWriter, UI};
+
+pub fn start(ui: &mut UI) -> Result<()> {
+    let dt = Local::now();
+    let tarball_name = format!("support-bundle-{}.tar.gz", dt.format("%Y%m%d%H%M%S"));
+    let cwd = env::current_dir().unwrap();
+
+    ui.status(
+        Status::Generating,
+        format!("New Support Bundle at {}", dt.format("%Y-%m-%d %H:%M:%S")),
+    )?;
+
+    let sup_root = Path::new(&*FS_ROOT_PATH).join("hab").join("sup");
+    let tar_gz = File::create(&tarball_name)?;
+    let enc = GzEncoder::new(tar_gz, Compression::default());
+    let mut tar = tar::Builder::new(enc);
+
+    tar.follow_symlinks(false);
+
+    if sup_root.exists() {
+        ui.status(
+            Status::Adding,
+            format!("files from {}", &sup_root.display()),
+        )?;
+        tar.append_dir_all(format!("hab{}sup", MAIN_SEPARATOR), &sup_root)?;
```

Comment:
If I don't have sufficient permission (specifically, to read the `/hab/sup/default/CTL_SECRET` file), the command apparently fails, but I end up with an incomplete tarball.

```sh
☛ Generating New Support Bundle at 2018-05-08 13:25:00
☛ Adding files from /hab/sup
✗✗✗
✗✗✗ Permission denied (os error 13)
✗✗✗
```
I'm missing the following files: 
```sh
hab/sup/default/CTL_SECRET
hab/sup/default/LOCK
hab/sup/default/MEMBER_ID
```
It looks like it just fails at the first file it can't read, keeping everything before that, but leaving out everything after that. It would be good to catch this and clean up so we don't leave potentially confusing partial tarballs lying around.

If I run the command with `sudo`, however, it all works as advertised.